src/mongocrypt-marking.c

```diff
@@ -916,6 +945,263 @@ _mongocrypt_fle2_placeholder_to_find_ciphertext (
+                  mc_bson_type_to_string (bsonType),
+                  mc_bson_type_to_string (bson_iter_type (&lowerBound)));
+      goto fail;
+   }
+
+   if (bson_iter_type (&upperBound) != bsonType) {
+      CLIENT_ERR ("expected upperBound to match index type %s, got %s",
+                  mc_bson_type_to_string (bsonType),
+                  mc_bson_type_to_string (bson_iter_type (&upperBound)));
+      goto fail;
+   }
+
+   switch (bsonType) {
+   case BSON_TYPE_INT32:
+      BSON_ASSERT (bson_iter_type (&lowerBound) == BSON_TYPE_INT32);
+      BSON_ASSERT (bson_iter_type (&upperBound) == BSON_TYPE_INT32);
+      BSON_ASSERT (bson_iter_type (&findSpec->indexMin) == BSON_TYPE_INT32);
+      BSON_ASSERT (bson_iter_type (&findSpec->indexMax) == BSON_TYPE_INT32);
+      ret = mc_getMincoverInt32 (
+         (mc_getMincoverInt32_args_t){
+            .lowerBound = bson_iter_int32 (&lowerBound),
+            .includeLowerBound = includeLowerBound,
+            .upperBound = bson_iter_int32 (&upperBound),
+            .includeUpperBound = includeUpperBound,
+            .min = OPT_I32 (bson_iter_int32 (&findSpec->indexMin)),
+            .max = OPT_I32 (bson_iter_int32 (&findSpec->indexMax)),
+            .sparsity = sparsity},
+         status);
+      break;
+
+   case BSON_TYPE_INT64:
+      BSON_ASSERT (bson_iter_type (&lowerBound) == BSON_TYPE_INT64);
+      BSON_ASSERT (bson_iter_type (&upperBound) == BSON_TYPE_INT64);
+      BSON_ASSERT (bson_iter_type (&findSpec->indexMin) == BSON_TYPE_INT64);
+      BSON_ASSERT (bson_iter_type (&findSpec->indexMax) == BSON_TYPE_INT64);
+      ret = mc_getMincoverInt64 (
+         (mc_getMincoverInt64_args_t){
+            .lowerBound = bson_iter_int64 (&lowerBound),
+            .includeLowerBound = includeLowerBound,
+            .upperBound = bson_iter_int64 (&upperBound),
+            .includeUpperBound = includeUpperBound,
+            .min = OPT_I64 (bson_iter_int64 (&findSpec->indexMin)),
+            .max = OPT_I64 (bson_iter_int64 (&findSpec->indexMax)),
+            .sparsity = sparsity},
+         status);
+      break;
+   case BSON_TYPE_DATE_TIME:
+   case BSON_TYPE_DOUBLE:
+   case BSON_TYPE_DECIMAL128:
+      CLIENT_ERR ("FLE2 find not yet implemented for type: %s",
+                  mc_bson_type_to_string (bsonType));
+      goto fail;
+
+   case BSON_TYPE_EOD:
+   case BSON_TYPE_UTF8:
+   case BSON_TYPE_DOCUMENT:
+   case BSON_TYPE_ARRAY:
+   case BSON_TYPE_BINARY:
+   case BSON_TYPE_UNDEFINED:
+   case BSON_TYPE_OID:
+   case BSON_TYPE_BOOL:
+   case BSON_TYPE_NULL:
+   case BSON_TYPE_REGEX:
+   case BSON_TYPE_DBPOINTER:
+   case BSON_TYPE_CODE:
+   case BSON_TYPE_SYMBOL:
+   case BSON_TYPE_CODEWSCOPE:
+   case BSON_TYPE_TIMESTAMP:
+   case BSON_TYPE_MAXKEY:
+   case BSON_TYPE_MINKEY:
+   default:
+      CLIENT_ERR ("FLE2 find is not supported for type: %s",
+                  mc_bson_type_to_string (bsonType));
+      goto fail;
+   }
+
+fail:
```

Comment:
If no cleanup handling is required for this function, suggest early return with `return false;` rather than `goto fail;`.